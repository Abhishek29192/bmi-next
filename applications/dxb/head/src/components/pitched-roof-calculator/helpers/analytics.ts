import { createContext } from "react";

export type OnAnalyticsEvent = (event: {
  event?: string;
  id: string;
  label?: string;
  action?: string;
}) => void;

export const AnalyticsContext = createContext<OnAnalyticsEvent>(() => {
  // no-op
});
