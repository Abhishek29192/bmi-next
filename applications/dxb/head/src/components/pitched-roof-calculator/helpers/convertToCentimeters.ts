import type { ClassificationField, Distance } from "@bmi/elasticsearch-types";

export function convertToCentimeters(input: Distance): number;
export function convertToCentimeters(
  input?: ClassificationField[0],
  throwOnTypeError?: boolean
): number | undefined;
export function convertToCentimeters(
  input?: ClassificationField[0] | Distance,
  throwOnTypeError = true
) {
  if (!input) {
    return;
  }

  const numberValue = Number(input.value);
  if (!numberValue && numberValue !== 0) {
    if (throwOnTypeError) {
      throw new Error("Value is not a number");
    }

    return;
  }

  //cm, mm, m, μm
  const unitOfLength =
    "unit" in input
      ? input.unit
      : input.code.slice(input.value.length).toLowerCase().trim();
  switch (unitOfLength) {
    case "mm":
      return numberValue / 10;
    case "m":
      return numberValue * 100;
    case "μm":
      return numberValue / 10000;
    default:
      return numberValue;
  }
}
