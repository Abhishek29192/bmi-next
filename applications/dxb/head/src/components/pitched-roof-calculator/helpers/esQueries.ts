import { ProductReference } from "@bmi/elasticsearch-types";
import { ProductType } from "../types";

const sharedESRules = [
  { match: { "CHANNEL.code": "CALCULATOR" } },
  { exists: { field: "externalProductCode" } },
  { exists: { field: "name" } }
];

export const getProductsQuery = (pitches: number[]) => ({
  size: 100,
  query: {
    bool: {
      should: [
        {
          bool: {
            must: [
              ...sharedESRules,
              {
                match: {
                  "GENERALINFORMATION$PRODUCTTYPE.code": ProductType.tile
                }
              },
              { exists: { field: "MEASUREMENTS$WIDTH" } },
              { exists: { field: "MEASUREMENTS$LENGTH" } },
              { exists: { field: "TILESATTRIBUTES$MINIMUMBATTENSPACING" } },
              { exists: { field: "TILESATTRIBUTES$RIDGESPACE" } },
              { exists: { field: "GENERALINFORMATION$CLASSIFICATION" } },
              { exists: { field: "APPEARANCEATTRIBUTES$COLOUR" } }
            ],
            filter: pitches.flatMap((pitchValue) => [
              {
                range: {
                  "battenSpacings.maxAngle": {
                    gte: pitchValue
                  }
                }
              },
              {
                range: {
                  "battenSpacings.minAngle": {
                    lte: pitchValue
                  }
                }
              },
              {
                range: {
                  "TILESATTRIBUTES$RIDGESPACEENDANGLE.value": {
                    gte: pitchValue,
                    lt: 90
                  }
                }
              },
              {
                range: {
                  "TILESATTRIBUTES$RIDGESPACESTARTANGLE.value": {
                    lte: pitchValue
                  }
                }
              }
            ])
          }
        },
        {
          bool: {
            must: [
              ...sharedESRules,
              {
                match: {
                  "GENERALINFORMATION$PRODUCTTYPE.code": ProductType.underlay
                }
              },
              { exists: { field: "MEASUREMENTS$WIDTH" } },
              { exists: { field: "MEASUREMENTS$LENGTH" } },
              { exists: { field: "UNDERLAYATTRIBUTES$OVERLAP" } }
            ],
            filter: pitches.map((pitch) => ({
              range: {
                "UNDERLAYATTRIBUTES$MINSUPPORTEDPITCH.value": {
                  lte: pitch
                }
              }
            }))
          }
        },
        {
          bool: {
            must: [
              ...sharedESRules,
              {
                terms: {
                  "GENERALINFORMATION$PRODUCTTYPE.code.keyword": [
                    ProductType.gutter,
                    ProductType.gutterHook
                  ]
                }
              },
              { exists: { field: "MEASUREMENTS$LENGTH" } }
            ]
          }
        }
      ]
    }
  }
});

export const constructQueryForProductReferences = (
  productReferences: ProductReference[]
) => ({
  size: 50,
  query: {
    bool: {
      must: [
        {
          terms: {
            "code.keyword": productReferences.map(
              (productReference) => productReference.code
            )
          }
        },
        { exists: { field: "externalProductCode" } },
        { exists: { field: "name" } }
      ]
    }
  }
});
