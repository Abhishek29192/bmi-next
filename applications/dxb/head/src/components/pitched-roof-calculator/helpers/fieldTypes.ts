const isPositiveNumber = (value: string) => Number.parseFloat(value) > 0;

/*
  TODO: messages to be converted to using microcopy-like replacement
  https://bmigroup.atlassian.net/browse/WEBT-202
 */

type ErrorType = "fieldRequired" | "positiveNumberOnly" | "range";

type GetErrorMessage = (
  errorType: ErrorType,
  placeholders?: Record<string, string>
) => string;

const numberValidator = (value?: string, errorMessage?: GetErrorMessage) => {
  if (!value) {
    return errorMessage
      ? errorMessage("fieldRequired")
      : "This field is required";
  }

  if (!isPositiveNumber(value)) {
    return errorMessage
      ? errorMessage("positiveNumberOnly")
      : "This field must have a positive number";
  }
};

type RangeValidatorCreator = (
  min: number,
  max: number,
  getErrorMessage?: GetErrorMessage
) => (value?: string) => string | undefined;

const rangeValidator: RangeValidatorCreator =
  (min, max, getErrorMessage) => (value) => {
    const error = numberValidator(value, getErrorMessage);
    if (error) return error;

    const number = value ? parseFloat(value) : undefined;

    if (!number || number < min || number > max) {
      return getErrorMessage
        ? getErrorMessage("range", { min: min.toString(), max: max.toString() })
        : `This number must be between ${min} and ${max}`;
    }
  };

export type Type = "LENGTH" | "PROTRUSION_LENGTH" | "PITCH";

type TypeProps = {
  unit: string;
  validator: (value?: string) => string | undefined;
  helperText?: string;
};

export type FieldTypesPath = "lengthRange" | "pitchRange" | ErrorType;

export const getFieldTypes = (
  getMicroCopy: (
    path: FieldTypesPath,
    placeholders?: Record<string, string>
  ) => string
): {
  [key in Type]: TypeProps;
} => ({
  LENGTH: {
    unit: "m",
    validator: rangeValidator(2, 200, (path, placeholders) =>
      path === "range"
        ? getMicroCopy("lengthRange", placeholders)
        : getMicroCopy(path, placeholders)
    ),
    helperText: getMicroCopy("lengthRange", { min: "2", max: "200" })
  },
  PROTRUSION_LENGTH: {
    unit: "m",
    validator: rangeValidator(0.5, 200, (path, placeholders) =>
      path === "range"
        ? getMicroCopy("lengthRange", placeholders || {})
        : getMicroCopy(path, placeholders || {})
    ),
    helperText: getMicroCopy("lengthRange", { min: "0.5", max: "200" })
  },
  PITCH: {
    unit: "deg",
    validator: rangeValidator(6, 85, (path, placeholders) =>
      path === "range"
        ? getMicroCopy("pitchRange", placeholders)
        : getMicroCopy(path, placeholders)
    ),
    helperText: getMicroCopy("pitchRange", { min: "6", max: "85" })
  }
});
