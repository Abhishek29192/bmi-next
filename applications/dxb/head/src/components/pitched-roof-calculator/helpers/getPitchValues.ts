import type { DimensionsValues } from "../types/roof";

const getPitchValues = ({ protrusions, ...dimensions }: DimensionsValues) => {
  const mainRoofPitchValues = getPitchFromDimensions(dimensions);

  const protrusionPitchValues =
    protrusions?.flatMap(getPitchFromDimensions) || [];

  return [...mainRoofPitchValues, ...protrusionPitchValues];
};

const getPitchFromDimensions = (dimensions: DimensionsValues) =>
  Object.keys(dimensions)
    .filter((key) => key.startsWith("P"))
    // eslint-disable-next-line security/detect-object-injection
    .map((key) => parseFloat(dimensions[key])); // All items which start with P are roof pitch values (e.g. P, P1, P2, etc...)

export default getPitchValues;
