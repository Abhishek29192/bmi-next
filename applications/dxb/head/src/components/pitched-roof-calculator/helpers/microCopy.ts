export { MicroCopy, MicroCopyContext, getMicroCopy } from "@bmi/components";
