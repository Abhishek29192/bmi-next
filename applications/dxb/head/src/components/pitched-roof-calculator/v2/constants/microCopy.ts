export const microCopy = {
  ROOF_SELECTION_TITLE: "roofSelection.title",
  ROOF_SELECTION_SUBTITLE: "roofSelection.subtitle",
  ROOF_SELECTION_GABLE: "roofSelection.gable",
  ROOF_SELECTION_HIPPED: "roofSelection.hipped",
  ROOF_SELECTION_SLOPED: "roofSelection.sloped",
  ROOF_SELECTION_ROOF: "roofSelection.roof",
  ROOF_DIMENSIONS_TITLE: "roofDimensions.title",
  ROOF_DIMENSIONS_SUBTITLE: "roofDimensions.subtitle",
  ROOF_DIMENSIONS_NEXT_LABEL: "roofDimensions.nextLabel",
  ROOF_DIMENSIONS_BACK_LABEL: "roofDimensions.backLabel",
  ROOF_DIMENSIONS_PROTRUSIONS_TITLE: "roofDimensions.protrusions.title",
  ROOF_DIMENSIONS_PROTRUSIONS_ADD: "roofDimensions.protrusions.add",
  ROOF_DIMENSIONS_PROTRUSIONS_PROMPT: "roofDimensions.protrusions.prompt",
  ROOF_DIMENSIONS_PROTRUSIONS_ADD_ANOTHER:
    "roofDimensions.protrusions.addAnother",
  ROOF_DIMENSIONS_PROTRUSIONS_REMOVE: "roofDimensions.protrusions.remove",
  TILE_SELECTION_TITLE: "tileSelection.title",
  TILE_SELECTION_SUBTITLE: "tileSelection.subtitle",
  TILE_SELECTION_BACK_LABEL: "tileSelection.backLabel",
  TILE_SELECTION_CATEGORIES_CONCRETE: "tileSelection.categories.concrete",
  TILE_SELECTION_CATEGORIES_CLAY: "tileSelection.categories.clay",
  TILE_SELECTION_CATEGORIES_METAL: "tileSelection.categories.metal",
  TILE_SELECTION_COLOR: "tileSelection.color",
  TILE_SELECTION_COLORS: "tileSelection.colors",
  TILE_SELECTION_EMPTY: "tileSelection.empty",
  VARIANT_SELECTION_TITLE: "variantSelection.title",
  VARIANT_SELECTION_SUBTITLE: "variantSelection.subtitle",
  VARIANT_SELECTION_BACK_LABEL: "variantSelection.backLabel",
  TILE_OPTIONS_TITLE: "tileOptions.title",
  TILE_OPTIONS_SUBTITLE: "tileOptions.subtitle",
  TILE_OPTIONS_NEXT_LABEL: "tileOptions.nextLabel",
  TILE_OPTIONS_BACK_LABEL: "tileOptions.backLabel",
  TILE_OPTIONS_VERGE_TITLE: "tileOptions.verge.title",
  TILE_OPTIONS_VERGE_HELP: "tileOptions.verge.help",
  TILE_OPTIONS_VERGE_NONE_LABEL: "tileOptions.verge.noneLabel",
  TILE_OPTIONS_RIDGE_TITLE: "tileOptions.ridge.title",
  TILE_OPTIONS_RIDGE_HELP: "tileOptions.ridge.help",
  TILE_OPTIONS_VENTILATION_HOOD_TITLE: "tileOptions.ventilationHood.title",
  TILE_OPTIONS_VENTILATION_HOOD_HELP: "tileOptions.ventilationHood.help",
  TILE_OPTIONS_VENTILATION_HOOD_NONE_LABEL:
    "tileOptions.ventilationHood.noneLabel",
  UNDERLAY_SELECTION_TITLE: "underlaySelection.title",
  UNDERLAY_SELECTION_SUBTITLE: "underlaySelection.subtitle",
  UNDERLAY_SELECTION_NEXT_LABEL: "underlaySelection.nextLabel",
  UNDERLAY_SELECTION_BACK_LABEL: "underlaySelection.backLabel",
  UNDERLAY_SELECTION_EMPTY: "underlaySelection.empty",
  GUTTERING_TITLE: "guttering.title",
  GUTTERING_SUBTITLE: "guttering.subtitle",
  GUTTERING_NEXT_LABEL: "guttering.nextLabel",
  GUTTERING_BACK_LABEL: "guttering.backLabel",
  GUTTERING_SKIP_LABEL: "guttering.skipLabel",
  GUTTERING_GUTTER_TITLE: "guttering.gutter.title",
  GUTTERING_GUTTER_VARIANT_TITLE: "guttering.gutterVariant.title",
  GUTTERING_GUTTER_HOOK_TITLE: "guttering.gutterHook.title",
  GUTTERING_DOWN_PIPE_TITLE: "guttering.downPipe.title",
  GUTTERING_DOWN_PIPE_CONNECTORS_TITLE: "guttering.downPipeConnectors.title",
  RESULTS_TITLE: "results.title",
  RESULTS_SUBTITLE: "results.subtitle",
  RESULTS_AREA_LABEL: "results.areaLabel",
  RESULTS_BACK_LABEL: "results.backLabel",
  RESULTS_START_OVER_LABEL: "results.startOverLabel",
  RESULTS_DOWNLOAD_PDF_LABEL: "results.downloadPdfLabel",
  RESULTS_CATEGORIES_TITLES: "results.categories.tiles",
  RESULTS_CATEGORIES_FIXINGS: "results.categories.fixings",
  RESULTS_CATEGORIES_SEALING: "results.categories.sealing",
  RESULTS_CATEGORIES_VENTILATION: "results.categories.ventilation",
  RESULTS_CATEGORIES_ACCESSORIES: "results.categories.accessories",
  RESULTS_TABLE_TITLE: "results.table.title",
  RESULTS_TABLE_PACK_SIZE: "results.table.packSize",
  RESULTS_TABLE_EXTERNAL_PRODUCT_CODE: "results.table.externalProductCode",
  RESULTS_TABLE_QUANTITY: "results.table.quantity",
  RESULTS_TABLE_REMOVE: "results.table.remove",
  RESULTS_ALERTS_QUANTITIES_NOT_UPDATED_TITLE:
    "results.alerts.quantities.not.updated.title",
  RESULTS_ALERTS_QUANTITIES_NOT_UPDATED_TEXT:
    "results.alerts.quantities.not.updated.text",
  RESULTS_ALERTS_QUANTITIES_UPDATED_TITLE:
    "results.alerts.quantities.updated.title",
  RESULTS_ALERTS_QUANTITIES_UPDATED_TEXT:
    "results.alerts.quantities.updated.text",
  RESULTS_ALERTS_QUANTITIES_RESET_BUTTON:
    "results.alerts.quantities.reset.button",
  RESULTS_ALERTS_NEED_TO_KNOW_TITLE: "results.alerts.needToKnow.title",
  RESULTS_ALERTS_NEED_TO_KNOW_TEXT: "results.alerts.needToKnow.text",
  RESULTS_EMAIL_TITLE: "results.email.title",
  RESULTS_EMAIL_HELP: "results.email.help",
  RESULTS_EDITED_PRODUCTS_TITLE: "results.edited.products.title",
  PDF_REPORT_EXTRAS_SECTION_TITLE: "pdf.report.extras.section.title",
  VALIDATION_ERRORS_FIELD_REQUIRED: "validation.errors.fieldRequired",
  VALIDATION_ERRORS_POSITIVE_NUMBERS_ONLY:
    "validation.errors.positiveNumberOnly",
  VALIDATION_ERRORS_LENGTH_RANGE: "validation.errors.lengthRange",
  VALIDATION_PITCH_RANGE: "validation.errors.pitchRange",
  CALCULATOR_NOBB_LABEL: "calculator.nobb.label"
};
