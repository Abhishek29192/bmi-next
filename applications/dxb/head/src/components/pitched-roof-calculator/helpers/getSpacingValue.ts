import type { BattenSpacing } from "@bmi/elasticsearch-types";
import { convertToCentimeters } from "./convertToCentimeters";

export const getSpacingByPitch = (
  battenSpacings: BattenSpacing[],
  pitch: number
): { eaveGauge: number; maxBattenSpacing: number } => {
  const battenSpacing = battenSpacings.find(
    (batten) => batten.maxAngle >= pitch && batten.minAngle <= pitch
  );

  return {
    eaveGauge: convertToCentimeters(battenSpacing!.firstRowBattenDistance),
    maxBattenSpacing: convertToCentimeters(battenSpacing!.battenDistance)
  };
};
