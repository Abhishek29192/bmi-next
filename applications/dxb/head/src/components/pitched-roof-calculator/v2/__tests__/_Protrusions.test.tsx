import { FormContext } from "@bmi/components";
import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { MicroCopy } from "../../helpers/microCopy";
import en from "../../samples/copy/en.json";
import Protrusions from "../_Protrusions";

describe("PitchedRoofCalculator Protrusions component", () => {
  it("renders correctly", () => {
    const updateFormState = jest.fn();
    const hasBeenSubmitted = false;
    const submitButtonDisabled = false;

    const { container } = render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState,
            hasBeenSubmitted,
            submitButtonDisabled,
            values: {}
          }}
        >
          <Protrusions />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("adds a protrusion", () => {
    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: jest.fn(),
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <Protrusions
            defaultValue={[{ type: "protrusion01", A: 3, B: 3, P: 30 }]}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    const addProtrusionButton = screen.getByText(
      "MC: roofDimensions.protrusions.addAnother"
    );
    fireEvent.click(addProtrusionButton);
    expect(
      screen.getAllByText("MC: roofDimensions.protrusions.remove").length
    ).toBe(2);
  });

  it("renders without 'Add another button' if there are more then 8 protrusions", () => {
    const protrusion = { type: "protrusion01", A: 3, B: 3, P: 30 };

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: jest.fn(),
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <Protrusions
            defaultValue={[
              protrusion,
              protrusion,
              protrusion,
              protrusion,
              protrusion,
              protrusion,
              protrusion,
              protrusion,
              protrusion
            ]}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    expect(
      screen.queryByText("MC: roofDimensions.protrusions.addAnother")
    ).not.toBeInTheDocument();
  });

  it("removes all protrusions", () => {
    const updateFormState = jest.fn();
    const hasBeenSubmitted = false;
    const submitButtonDisabled = false;

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState,
            hasBeenSubmitted,
            submitButtonDisabled,
            values: {}
          }}
        >
          <Protrusions
            defaultValue={[
              {},
              { type: "protrusion01", A: "33", B: "55", P: "44" }
            ]}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    const removeProtrusionButtons = screen.getAllByText(
      "MC: roofDimensions.protrusions.remove"
    );
    removeProtrusionButtons.forEach((button) => {
      fireEvent.click(button);
    });

    expect(
      screen.queryByText("MC: roofDimensions.protrusions.remove")
    ).not.toBeInTheDocument();
  });
});
