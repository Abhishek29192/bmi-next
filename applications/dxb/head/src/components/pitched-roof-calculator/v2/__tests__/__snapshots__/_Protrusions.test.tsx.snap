// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PitchedRoofCalculator Protrusions component renders correctly 1`] = `
<div>
  <div
    class="Protrusions"
  >
    <h6
      class="title"
    >
      MC: roofDimensions.protrusions.title
    </h6>
    <div
      class="protrusion"
    >
      <div
        class="FieldContainer fieldsContainer"
      >
        <div
          class="MuiGrid-root Grid Grid--justifyContent MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
        >
          <div
            class="MuiGrid-root card MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput"
            >
              <label>
                <input
                  class="input"
                  name="select-protrusion-0"
                  type="radio"
                  value="protrusion01"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <svg
                      preserveAspectRatio="xMidYMid"
                    />
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root card MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput"
            >
              <label>
                <input
                  class="input"
                  name="select-protrusion-0"
                  type="radio"
                  value="protrusion02"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <svg
                      preserveAspectRatio="xMidYMid"
                    />
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root card MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput"
            >
              <label>
                <input
                  class="input"
                  name="select-protrusion-0"
                  type="radio"
                  value="protrusion03"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <svg
                      preserveAspectRatio="xMidYMid"
                    />
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
